src/Aspire.Cli/Commands/ExecCommand.cs

```diff
@@ -0,0 +1,307 @@
+        {
+            _interactionService.DisplayError(InteractionServiceStrings.ProjectOptionNotSpecifiedNoCsprojFound);
+            return ExitCodeConstants.FailedToFindProject;
+        }
+        catch (AppHostIncompatibleException ex)
+        {
+            return _interactionService.DisplayIncompatibleVersionError(
+                ex,
+                appHostCompatibilityCheck?.AspireHostingVersion ?? throw new InvalidOperationException(ErrorStrings.AspireHostingVersionNull)
+                );
+        }
+        catch (CertificateServiceException ex)
+        {
+            _interactionService.DisplayError(string.Format(CultureInfo.CurrentCulture, TemplatingStrings.CertificateTrustError, ex.Message));
+            return ExitCodeConstants.FailedToTrustCertificates;
+        }
+        catch (FailedToConnectBackchannelConnection ex)
+        {
+            _interactionService.DisplayError(string.Format(CultureInfo.CurrentCulture, InteractionServiceStrings.ErrorConnectingToAppHost, ex.Message));
+            _interactionService.DisplayLines(runOutputCollector.GetLines());
+            return ExitCodeConstants.FailedToDotnetRunAppHost;
+        }
+        catch (Exception ex)
+        {
+            _interactionService.DisplayError(string.Format(CultureInfo.CurrentCulture, InteractionServiceStrings.UnexpectedErrorOccurred, ex.Message));
+            _interactionService.DisplayLines(runOutputCollector.GetLines());
+            return ExitCodeConstants.FailedToDotnetRunAppHost;
+        }
+    }
+
+    private (ICollection<string> arbitary, ICollection<string> command) ParseCmdArgs(ParseResult parseResult)
+    {
+        var allTokens = parseResult.UnmatchedTokens.ToList();
+        int delimiterIndex = allTokens.IndexOf("--");
+        List<string> arbitraryFlags = new();
+        List<string> commandTokens = new();
+
+        // Find the index of the first token that is not an option (doesn't start with '-') and is not the value for a known option
+        // We'll use the options defined in this command to skip known option values
+        var knownOptions = new HashSet<string>(Options.SelectMany(o => o.Aliases));
+        int i = 0;
+        while (i < allTokens.Count)
+        {
+            if (delimiterIndex >= 0 && i == delimiterIndex)
+            {
+                // Everything after -- is command
+                commandTokens.AddRange(allTokens.Skip(i + 1));
+                break;
+            }
+
+            var token = allTokens[i];
+            if (knownOptions.Contains(token))
+            {
+                // Skip the option and its value (if it has one)
+                var option = Options.FirstOrDefault(o => o.Aliases.Contains(token));
+                if (option is not null)
+                {
+                    // If the option is not a bool, it expects a value
+                    var optionType = option.GetType();
```

Comment:
Not sure if this is going to work with AOT.

Reply:
changed to not using `System.Type` anywhere